src/app/components/ElectronicDeliveryForm/ElectronicDeliveryForm.jsx

```diff
@@ -17,13 +17,17 @@ class ElectronicDeliveryForm extends React.Component {
     // in the no-js scenario. If they're not available, then we use this
     // 'fallback', but the empty object structure is needed.
     this.state = {
-      form: !_isEmpty(this.props.form) ? this.props.form :
-        {
-          emailAddress: this.props.defaultEmail,
-          chapterTitle: '',
-          startPage: '',
-          endPage: '',
-        },
+      form: !_isEmpty(this.props.form)
+        ? this.props.form
+        : // Use Formstate session data if it exists
+        !_isEmpty(window.sessionStorage.getItem('formstate'))
```

Comment:
Getting an error rendering this server-side - I think because of the `window` reference?

Reply:
Also, we may want to key the formstate to the specific item so that if someone starts to request an item and aborts for some reason, we don't prefill the EDD request form with the same data when they start to request a different item a week later, for example.

Reply:
> Getting an error rendering this server-side - I think because of the `window` reference?

Oops. Forgot to check window type.

Reply:
I understand now! Thanks for the explanation